### PR TITLE
Fix to avoid buffer overflow.

### PR DESCRIPTION
**Pull request by mikecaat**
Now, the buffer overflow which this patch handles doesn't happen
without this patch. But, it need to avoid the buffer overflow
even if the output message is changed in the future.

### CODE REVIEW

**Comment by mikecaat** [comment]
Note: this patch must be backpatched to other branches.

**Comment by MoonInsung** [comment]
Dear mikecaat.

Thank you for responding to this issue!!

In your opinion, we've been using the strncat wrong so far.
Before running strncat, we have already inputted the current date in the char array, so we need to modify it according to your opinion.
There has not been a bit of a problem, but I agree with your opinion, because it may display warnings depending on the version of GCC.

By the way, there is one more thing I want to ask for.
It would be okay to respond with a separate PR, but if possible, is it respond to some warnings currently occurring together?
When checking the CI, the following warning occurs.
```
https://github.com/ossc-db/pg_rman/runs/2434225435?check_suite_focus=true

data.c: In function ‘figure_out_segno’:
data.c:1267:8: warning: variable ‘scanned’ set but not used [-Wunused-but-set-variable]
 1267 |   int  scanned;
--
data.c: In function ‘restore_data_file’:
data.c:772:4: warning: ignoring return value of ‘truncate’, declared with attribute warn_unused_result [-Wunused-result]
  772 |    truncate(to_path, (blknum - 1) * BLCKSZ);
```

In other words, there are variable that is unused, and functions(truncate) have not responded to return value.
Is it possible to cope with this problem?

Best regards.
Moon.

**Comment by mikecaat** [comment]
Dear @MoonInsung 

Thanks for your response!

> It would be okay to respond with a separate PR, but if possible, is it respond to some warnings currently occurring together?
When checking the CI, the following warning occurs.

OK, I'll try.  I don't mind either to separate the patch or not.

It seems that the first warning seems to be used for the assertion error. 
I'll research how to handle the warning is best.

**Comment by MoonInsung** [comment]
Dear mikecaat.

Oops, Sorry.
This PR is a buffer overflow issue, so it is different from the one I asked for additionally.
I thought of fixing all warning messages, but this is a completely different comment from the original PR title.
Therefore, if there are no comments on the PR you sent, I will merge in after 4 hours.

Best regards.
Moon.

**Comment by mikecaat** [comment]
Dear @MoonInsung

OK. I understood.

BTW, the first assertion warning has occurred because of to build without --enable-cassert. So, I change in the [PR](https://github.com/ossc-db/pg_rman/pull/159)
The second warning has occurred because of to build with CFLAGS=-O2. We can make that quiet with CFALGS=-O0, but  I think it's not a good solution. So, I'll change the pg_rman's code.

Regards,

**Comment by MoonInsung** [comment]
It was merged.
I will use the commit ID to apply it to other branches.

**Comment by MoonInsung** [comment]
done.
